src/Form/grid/StyledContainer/hooks.ts

```diff
@@ -197,16 +214,19 @@ export const useContainerEngine = (node: any, rawNode: any, ref: any) => {
 
     // If the container is a fit container, we need to alter it's max-width to allow children to expand accordingly
     if (isFitContainer(div) && children.length > 0) {
-      const parentFitContainers = getParentFitContainers(div);
-
-      if (parentFitContainers) {
-        parentFitContainers.expand();
-      }
-
-      resizeFitContainer(div);
-
-      if (parentFitContainers) {
-        parentFitContainers.collapse();
+      resizeCurrentFitContainer();
+
+      // For the editor, we need to also resize after transitions finish (between viewport changes)
+      if (node.uuid) {
+        canvas = document.querySelector('div[data-testid="editor-canvas"]');
```

Comment:
is there any way all of hte editor-related logic can be passed in from the editor via a prop? maybe have the editor pass in a hook or something so we don't create editor dependencies in the react library (e.g. assuming the editor-canvas attribute here)

Reply:
I feel like it would be difficult to do due to some of the modifications of the editor vs. hosted form containers being so deep within logic. It's just slight changes.

I can start thinking about how we can navigate towards this approach if you want though.